### DISCUSSION TITLE
Choose backup location

### DISCUSSION THREAD

**Discussion by holocronweaver** [iSoron/uhabits#1912] [Feature Requests]
Currently the backup location for automatic backups is [hardcoded](https://github.com/iSoron/uhabits/commit/dc0b8deccf40be747fc2b34d7c836e4808146c51#diff-91f73f4b938e8bb1a8f5a30d545d3c5e2a430438e03e051e8d4db859f2b5a034R31) to (on most Android setups) `/storage/Android/data/org.isoron.uhabits/files/Backups/`.

This location works fine if you are using Google backup, but if you want to use another backup solution and do not have root then this location will be inaccessible.

To support alternate backup solutions I propose making this location user configurable under a new settings section, `Backups`, which will merge with the `Database` section so everything related to backups is in one place. Adding this setting has the added benefit of surfacing to users that automatic backups are being made, which I didn't realize until stumbling upon [the automatic backups feature request](https://github.com/iSoron/uhabits/discussions/178).

I'm happy to implement this feature myself, but wanted to gain some feedback from maintainers and users before proceeding.

**Comment by jruiz94** [comment]
Yes I really miss this feature! I really wanted to backup my habits using Syncthing just in case I lose the phone!

**Comment by kaoneko** [comment]
Yes, this is sorely needed now that Google has turned the shared app storage into a second private app storage.

**Comment by 0ldb34r** [comment]
I use MiX as my file manager and when the backup file is created, the app offers me to choose which file I want to open it with. By choosing MiX, I systematically select the folder where I want it to be saved. Pending a possible fix, this may help some people.

**Reply by kaoneko** [reply]
Are you saying that after doing that once the automatic backups are also processed by MiX or something?

**Reply by 0ldb34r** [reply]
Yes, I can define MiX as the default app for treating these backups and then it puts them in the folser i chose.

**Reply by kaoneko** [reply]
That's pretty wild. I use Total Commander to share manual backups to, but automated backups are still saved in the folder mentioned by OP.

**Comment by krwow** [comment]
Saving to local storage is also possible using "Save Locally" app from Play store or "Shoyo" from GitHub, they are both free, but that's still a workaround, not a solution to this issue. I have no other app which lacks such local saving feature, I have to use one of the mentioned before apps just for this habits app.

@iSoron , any chances to add this feature, please?
